web/src/pages/api/public/events.ts

```diff
@@ -1,28 +1,44 @@
 import {
   PostEventsV1Body,
   PostEventsV1Response,
-  transformEventToIngestionBatch,
 } from "@/src/features/public-api/types/events";
 import { withMiddlewares } from "@/src/features/public-api/server/withMiddlewares";
 import { createAuthedAPIRoute } from "@/src/features/public-api/server/createAuthedAPIRoute";
-import { parseSingleTypedIngestionApiResponse } from "@/src/pages/api/public/ingestion";
-import { handleBatch } from "@langfuse/shared/src/server";
-import { tokenCount } from "@/src/features/ingest/usage";
+import { processEventBatch } from "@/src/pages/api/public/ingestion";
+import { eventTypes, logger } from "@langfuse/shared/src/server";
+import { v4 } from "uuid";
 
 export default withMiddlewares({
   POST: createAuthedAPIRoute({
     name: "Create Event",
     bodySchema: PostEventsV1Body,
     responseSchema: PostEventsV1Response,
-    fn: async ({ body, auth }) => {
-      const ingestionBatch = transformEventToIngestionBatch(body);
-      const result = await handleBatch(ingestionBatch, auth, tokenCount);
-      const response = parseSingleTypedIngestionApiResponse(
-        result.errors,
-        result.results,
-        PostEventsV1Response,
-      );
-      return response;
+    fn: async ({ body, auth, res }) => {
+      const event = {
+        id: v4(),
+        type: eventTypes.OBSERVATION_CREATE,
+        timestamp: new Date().toISOString(),
+        body: {
+          ...body,
+          type: "EVENT",
+        },
+      };
+      if (!event.body.id) {
+        event.body.id = v4();
+      }
+      const result = await processEventBatch([event], auth);
+      if (result.errors.length > 0) {
+        const error = result.errors[0];
+        res
+          .status(error.status)
+          .json({ message: error.error ?? error.message });
+        return { id: "" }; // dummy return
```

Comment:
why is this needed? Cant you just return res
          .status(error.status)
          .json({ message: error.error ?? error.message });